Assets/Scripts/Networking/PlayerManager.cs

```diff
@@ -239,21 +240,45 @@ private IEnumerator ReturnPlayerControl(float waitTime)
             }
         }
 
-        #endregion
+        private IEnumerator ShootPaintball()
+        {
+            waitingToShoot = true;
+           
+            GameObject paintball;
+            if (photonView.IsMine)  //We check if this is the local player shooting
+            {
+                paintball = Instantiate(paintballPrefab, paintGun.transform.position, Quaternion.identity);
+                paintball.GetComponent<Paintball>().playerWhoShot = this.gameObject;
+                paintball.GetComponent<Paintball>().paintballDamage = this.paintballDamage;
+            }
+            else //we spawn fake bullets for other players
+            {
+                paintball = Instantiate(paintballPrefabClient, paintGun.transform.position, Quaternion.identity);
+                paintball.GetComponent<PaintballClientSide>().playerWhoShot = this.gameObject;
+            }
+            
+            paintball.GetComponent<Rigidbody>().velocity = paintGun.TransformDirection(Vector3.forward * paintBallSpeed);
+            yield return new WaitForSeconds(shootWaitTime);
+            waitingToShoot = false;
+        }
+
+    #endregion
 
-        #region IPunObservable implementation
+    #region IPunObservable implementation
 
-        public void OnPhotonSerializeView(PhotonStream stream, PhotonMessageInfo info)
+    public void OnPhotonSerializeView(PhotonStream stream, PhotonMessageInfo info)
         {
             if (stream.IsWriting)
             {
                 // We own this player: send the others our data
                 stream.SendNext(this.health);
+                stream.SendNext(this.IsFiring);
             }
             else
             {
                 // Network player, receive data
                 this.health = (float)stream.ReceiveNext();
+                this.IsFiring = (bool)stream.ReceiveNext();
```

Comment:
we use the same variable for saying that we shoot and other player is shooting? It's a bit confusing. Maybe it could be separated?

Reply:
I think this is simple enough. Everything is the same except we shoot a different bullet depending on whether the player is local or not